compat/os/mswindows.h

```diff
@@ -404,7 +404,7 @@ typedef char * caddr_t;
 
 /* internal to Microsoft CRTLIB */
 typedef struct {
-    long osfhnd;    /* underlying OS file HANDLE */
+    HANDLE osfhnd;    /* underlying OS file HANDLE */
```

Comment:
For the code to work, our `osfhnd` type must match whatever type the actual `__pioinfo` array is using for that data member. FWIW, I cannot find authoritative information on that internal-to-Windows array and have no idea whether all Windows versions use `HANDLE` for the first data member of that array elements.

We should test this (which means not just compiling but _running_ this code and making sure osfhnd values in Windows memory make sense). If we cannot test this, then I doubt we should be fixing this problem because we would be just guessing a rather sensitive/critical type. We can easily make things worse. Converting this PR into a Draft PR and waiting for somebody who can help may be an overall better strategy if we cannot test.